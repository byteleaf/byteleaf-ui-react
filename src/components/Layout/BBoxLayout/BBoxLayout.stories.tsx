import React from 'react';
import Typography from '@material-ui/core/Typography';
import { BBoxLayout } from './BBoxLayout';

export default {
  title: 'Layout|BBoxLayout',
};

export const BBoxLayoutStorySmall = () => (
  <BBoxLayout size="small">
    <div style={{ backgroundColor: '#ddd', padding: '8px' }}>
      <Typography>This div is in a boxed layout.</Typography>
    </div>
  </BBoxLayout>
);

export const BBoxLayoutStoryMedium = () => (
  <BBoxLayout size="medium">
    <div style={{ backgroundColor: '#ddd', padding: '8px' }}>
      <Typography>This div is in a boxed layout.</Typography>
    </div>
  </BBoxLayout>
);

export const BBoxLayoutStoryLarge = () => (
  <BBoxLayout size="large">
    <div style={{ backgroundColor: '#ddd', padding: '8px' }}>
      <Typography>This div is in a boxed layout.</Typography>
    </div>
  </BBoxLayout>
);

BBoxLayoutStorySmall.story = {
  name: 'small',
  parameters: {
    notes: 'Small boxed layout. Will use half of the screen on large displays.',
  },
};

BBoxLayoutStoryMedium.story = {
  name: 'medium',
  parameters: {
    notes: 'Medium boxed layout. Will use half of the screen on large displays.',
  },
};

BBoxLayoutStoryLarge.story = {
  name: 'large',
  parameters: {
    notes: 'Large boxed layout. Will use half of the screen on large displays.',
  },
};
