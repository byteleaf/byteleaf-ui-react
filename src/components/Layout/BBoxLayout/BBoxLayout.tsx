import React from 'react';
import { Grid } from '@material-ui/core';

interface Sizes {
  [size: string]: {
    [resolution: string]: 12 | 10 | 8 | 6 | 4 | 2 | 1 | 'auto';
  };
}

const sizes: Sizes = {
  small: {
    xs: 12,
    sm: 8,
    md: 6,
    lg: 4,
  },
  medium: {
    xs: 12,
    sm: 12,
    md: 8,
    lg: 6,
  },
  large: {
    xs: 12,
    sm: 12,
    md: 10,
    lg: 8,
  },
};

export const BBoxLayout = ({
  children,
  size = 'medium',
}: {
  children: JSX.Element | JSX.Element[];
  size?: 'small' | 'medium' | 'large';
}) => (
  <Grid container justify="center">
    <Grid item xs={sizes[size].xs} sm={sizes[size].sm} md={sizes[size].md} lg={sizes[size].lg}>
      {children}
    </Grid>
  </Grid>
);
