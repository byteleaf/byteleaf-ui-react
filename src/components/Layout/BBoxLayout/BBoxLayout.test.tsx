import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { BBoxLayout } from './BBoxLayout';

describe('BBoxLayout', () => {
  it('renders correctly', async () => {
    const { container } = render(
      <BBoxLayout>
        <div>Layout</div>
      </BBoxLayout>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly on small', async () => {
    const { container } = render(
      <BBoxLayout size="small">
        <div>Layout</div>
      </BBoxLayout>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly on medium', async () => {
    const { container } = render(
      <BBoxLayout size="medium">
        <div>Layout</div>
      </BBoxLayout>,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly on large', async () => {
    const { container } = render(
      <BBoxLayout size="large">
        <div>Layout</div>
      </BBoxLayout>,
    );

    expect(container).toMatchSnapshot();
  });
});
