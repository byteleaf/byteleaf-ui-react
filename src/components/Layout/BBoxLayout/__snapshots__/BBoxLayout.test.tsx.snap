// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BBoxLayout renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-6"
    >
      <div>
        Layout
      </div>
    </div>
  </div>
</div>
`;

exports[`BBoxLayout renders correctly on large 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-10 MuiGrid-grid-lg-8"
    >
      <div>
        Layout
      </div>
    </div>
  </div>
</div>
`;

exports[`BBoxLayout renders correctly on medium 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-6"
    >
      <div>
        Layout
      </div>
    </div>
  </div>
</div>
`;

exports[`BBoxLayout renders correctly on small 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
    >
      <div>
        Layout
      </div>
    </div>
  </div>
</div>
`;
